src/eradiate/scenes/phase/_tabulated.py

```diff
@@ -95,15 +107,46 @@ def eval_mono(self, w: pint.Quantity) -> np.ndarray:
             Evaluated phase function as a 1D or 2D array depending on the shape
             of `w` (angle dimension comes last).
         """
-        return (
-            self.data.sel(i=0, j=0)
-            .interp(
-                w=w.m_as(self.data.w.units),
-                mu=np.linspace(-1, 1, self._n_mu),
+        # data already maps to regular grid of scattering angle cosines
+        if np.allclose(
+            self.data.mu.values,
+            np.linspace(-1.0, 1.0, self.data.mu.values.size),
+            rtol=1e-9,
+        ):
+            return (
+                self.data.sel(i=0, j=0)
+                .interp(
+                    w=w.m_as(self.data.w.units),
+                    kwargs=dict(bounds_error=True),
+                )
+                .data
+            )
+
+        # data does not map to regular grid of scattering angle cosines
+        else:
+            # compute the smallest regular grid that does not loose information
+            # in the sense of Shannon theorem
+            dmu = np.abs(self.data.mu.diff(dim="mu")).values.min()
+            nmu = 2 * int(np.ceil(2.0 / dmu))
```

Comment:
After discussion with @schunkes and @lucio-f, we could consider removing the outside factor 2 (from Shannon's theorem) which would result in lower memory footprint and faster interpolation

Reply:
Any thoughts about that? @leroyvn

Reply:
I'm not a signal processing expert so I don't have strong arguments to oppose: if you guys have proper arguments in favour of tuning down the number of grid points, go for it. Just keep that in mind when analysing results ;)

Reply:
Well Lucio made the point that the data is already sampled at a maximum frequency of dmu, so we don't really gain anything from resampling it at 2*dmu.

Reply:
For the record, my reasoning is that the smallest mu step is likely to be found at the extreme right of the [-1, 1] `mu` interval since this is where the forward scattering peak is located. Therefore, if we buid the regular `mu` grid using:

```python
dmu_min = np.abs(self.data.mu.diff(dim="mu")).values.min()
nmu = int(np.ceil(2.0 / dmu_min)) + 1
mu = np.linspace(-1, 1, nmu)
```
the two far-right mu points in [-1, 1] will exactly match the `mu` points in the input data. From these points on, the data to the left will be interpolated, therefore we incur a loss accuracy. Arguably, the loss of accuracy is minimised where it is important to be accuracy, namely close to `mu=1.0`.

Reply:
@schunkes I am not convinced that:

> dmu is the minimal mu step

implies that:

> 1 / dmu is the maximal frequency in the phase function signal

If you take the fft of a realistic particle phase function in the mu-space, you will find an infinite number of harmonics in my opinion

Reply:
After more discussion with the whiteboard, we found that assuming the following:
* the input data is propertly discretised, namely the smallest mu-step is found where the phase function has the sharpest variations
* the sharpest feature occurs either at `mu=-1` or at `mu=1.0`, or if it is located somewhere in the middle of the interval, the input data is well enough sampled in the `mu` space
the smallest delta mu regularisation method is the best.

As illustrated by the figure below, in the presumably most common case where the phase function has sharpest variations around mu=1.0, interpolating on the regular grid prescribed by the Shannon's theorem or on the regular grid using the smallest mu-step lead to the same result:

![phase_mu_interp](https://user-images.githubusercontent.com/62285993/169530086-f3325627-549e-4e02-a3a7-eaa3a61e0849.png)

Reply:
As a rule of thumb, the smallest the mu step in the input data, the lowest the interpolation errors.

We note that this regular grid tabulated phase function approach is not best suited for phase function with very sharp features, generally speaking, because of the overhead in memory and in interpolation time.

Reply:
However, this approach is the only available to us at the moment...